client/src/api/v1/index.ts

```diff
@@ -7,28 +11,55 @@ export default class ServerApi {
     this.api = axios.create({ baseURL: `api/v1` });
   }
 
-  protected async post<Type>(path: string, form: Type) {
+  protected async post<T, R>(
+    path: string,
+    form: T
+  ): Promise<{ status: number; data?: R }> {
     try {
       const response = await this.api.post(path, form);
-      return { response: response };
+      return { status: response.status, data: response.data };
     } catch (error) {
       if (axios.isAxiosError(error)) {
-        return { error: error };
+        return {
+          status: error.response?.status as number,
+          data: error.response?.data,
+        };
       }
+      throw new ApiError(`Could not post to path: ${path}`);
     }
   }
 
+  async createPost(form: {
+    title: string;
+    body: string;
+    file: string;
+    tags: string;
+    capacity: string;
+    location: string;
+  }) {
+    return await this.post<typeof form, { result?: Post; message?: string }>(
```

Comment:
I think the response type for this is: ``data: { result?: Post; message?: string }`` and not ``{ result?: Post; message?: string }``... just check to make sure.

When I was working on my API calls it was the former. In that case, you'll need to update any API calls that use the response data to be res.data.data.result instead of res.data.result